### PR TITLE
Migrate to `@vercel/style-guide`

### PR DESCRIPTION
**Pull request by mrmckeb**
This PR moves the configs over to use our (now public) style guide.

### CODE REVIEW

**Review by leerob — APPROVED** [review]
SGTM

**Comment by leerob** [comment]
Looks like there's some failing tests.

File: src/index.test.ts [hunk 1]
```diff
@@ -52,6 +52,8 @@ describe('ms(string)', () => {
     expect(isNaN(ms('☃'))).toBe(true);
     // @ts-expect-error - We expect this to fail.
     expect(isNaN(ms('10-.5'))).toBe(true);
+    // @ts-expect-error - We expect this to fail.
+    expect(isNaN(ms('ms'))).toBe(true);
```
Comment (mrmckeb) [inline src/index.test.ts:56]: I added this test case as it would be expected behaviour.

File: src/index.ts [hunk 2]
```diff
@@ -73,7 +73,7 @@ function msFn(value: StringValue | number, options?: Options): number | string {
   } catch (error) {
     const message = isError(error)
       ? `${error.message}. value=${JSON.stringify(value)}`
-      : 'An unknown error has occured.';
+      : 'An unknown error has occurred.';
```
Comment (mrmckeb) [inline src/index.ts:76]: That might have been me.

**Comment by mrmckeb** [comment]
Thanks @leerob, fixed. I blindly followed a lint suggestion without thinking about it - and it was a bad idea.

This commit fixes the issue, moving to named capture groups. I've checked browser/Node support and that seems good to go.
https://github.com/vercel/ms/pull/183/commits/2c17728fbf77461b37626d54c2a4afee6bde0905

**Review by CodyBrouwers — APPROVED** [review]
🚀

File: src/index.ts [hunk 3]
```diff
@@ -60,9 +60,9 @@ interface Options {
  * @param options - Options for the conversion
  * @throws Error if `value` is not a non-empty string or a number
  */
-function ms(value: StringValue, options?: Options): number;
-function ms(value: number, options?: Options): string;
-function ms(value: StringValue | number, options?: Options): number | string {
+function msFn(value: StringValue, options?: Options): number;
+function msFn(value: number, options?: Options): string;
+function msFn(value: StringValue | number, options?: Options): number | string {
```
Comment (styfle) [inline src/index.ts:65]: Why was `ms` changed to `msFn` here?

File: src/index.ts [hunk 4]
```diff
@@ -73,28 +73,31 @@ function ms(value: StringValue | number, options?: Options): number | string {
...
     const message = isError(error)
       ? `${error.message}. value=${JSON.stringify(value)}`
-      : 'An unknown error has occured.';
+      : 'An unknown error has occurred.';
     throw new Error(message);
   }
 }
 
 /**
  * Parse the given `str` and return milliseconds.
```
Comment (styfle) [inline src/index.ts:82]: This comment no longer matches the param name

File: src/index.ts [hunk 5]
```diff
@@ -73,28 +73,31 @@ function ms(value: StringValue | number, options?: Options): number | string {
...
+  // Named capture groups need to be manually typed today.
+  // https://github.com/microsoft/TypeScript/issues/32098
+  const groups = match?.groups as { value: string; type?: string } | undefined;
+  if (!groups) {
     return NaN;
   }
-  const n = parseFloat(match[1]);
-  const type = (match[2] || 'ms').toLowerCase() as Lowercase<Unit>;
+  const type = (groups.type || 'ms').toLowerCase() as Lowercase<Unit>;
+  const n = parseFloat(groups.value);
```
[the remark this answers is not in the document]
Comment (mrmckeb) [reply src/index.ts:100]: I refactored this a little after making the initial changes, I can flip back.

File: src/index.ts (hunk 4 above)
Comment (mrmckeb) [reply src/index.ts:82]: I'll update this.

File: src/index.ts (hunk 3 above)
Comment (mrmckeb) [reply src/index.ts:65]: It was shadowed (we use `ms` as a parameter name).

**Comment by mrmckeb** [comment]
@styfle, I've made the suggested changes here, along with a few other TSDoc improvements.
https://github.com/vercel/ms/pull/183/commits/3c8340ac5e9c733e805e6f9b5ea0ee3dda5080fc

**Review by styfle — APPROVED** [review]

**Pull request merged**